crates/uv-python/src/discovery.rs

```diff
@@ -1476,8 +1476,12 @@ impl PythonRequest {
                 version_request.matches_interpreter(interpreter)
             }
             PythonRequest::Directory(directory) => {
-                // `sys.prefix` points to the venv root.
+                // `sys.prefix` points to the environment root or `sys.executable` is the same
                 is_same_executable(directory, interpreter.sys_prefix())
```

Comment:
Arguably, we could remove this first check. Mildly afraid of something breaking though.